### DISCUSSION TITLE
Export to CSV

### DISCUSSION THREAD

**Discussion by dannyvankooten** [ibericode/koko-analytics#217] [Ideas]
Either for importing to another site or to create custom reports off. For the latter, one currently has to use go make requests to the REST API endpoints.

**Comment by dannyvankooten** [comment]
I wonder what the format for the CSV export should look like, which depends on what people want to do with them?

Easiest would be just a dump of the internal database table format that Koko Analytics uses, but another option would be to do an export based on the way the data is shown on the dashboard?

**Comment by JosKlever** [comment]
I've got a client who asked for an export to CSV. He would like to select the specific period and export the results as shown on the dashboard.

Other clients I've talked to use it create an annual report  for their members, so additionally an export of the totals per month/quarter/year could be nice.

**Comment by wiegertschouten** [comment]
I'm replacing a client site which uses Koko Analytics with another one which also uses it. But I can't transfer the data visitor stats. It would be nice if it would be possible. 

For this use case, the import/export format doesn't matter too much I guess.

**Comment by dannyvankooten** [comment]
@wiegertschouten Do the two sites have matching post and page ID's? Because Koko Analytics internally uses post ID's to refer to pages since that saves on a lot of storage, but sadly that also means exporting and importing between two slightly differing websites becomes pretty much impossible.

If the two sites do match then you should be able to export and import the Koko Analytics tables using something like PHPMyAdmin or a raw mysqldump.

**Comment by wiegertschouten** [comment]
@dannyvankooten Thank you for the quick reply! It's a completely new site, so the IDs do not match. After posting my previous comment, I already took a dive in the database structure and came to the same conclusion. It's a bit unfortunate, especially from my client's perspective. But I understand the tradeoff from your side.

**Comment by dannyvankooten** [comment]
This feature is now available in Koko Analytics Pro.

![image](https://github.com/user-attachments/assets/bb1c8a57-8ef5-4852-a257-f7dde9a76769)
